civicrm_entity.module

```diff
@@ -121,3 +127,35 @@ function civicrm_entity_pseudoconstant_options(FieldStorageDefinitionInterface $
   $options = $civicrm_api->getOptions($entity_type->get('civicrm_entity'), $definition->getName());
   return $options;
 }
+
+/**
+ * Implements hook_preprocess().
+ */
+function template_preprocess_civicrm_entity(&$variables) {
+  // Add fields as content to template.
+  $variables += ['content' => []];
+  foreach (Element::children($variables['elements']) as $key) {
+    $variables['content'][$key] = $variables['elements'][$key];
+  }
+
+  // Add the view_mode to the template.
+  $variables['view_mode'] = $variables['elements']['#view_mode'];
+
+  // Add the bundle to the template.
+  $variables['entity_type'] = $variables['elements']['#entity_type'];
+}
+
+/**
+ * Implements hook_theme_suggestions_HOOK_alter().
+ */
+function civicrm_entity_theme_suggestions_civicrm_entity_alter(array &$suggestions, array $variables) {
+  $entity_type = $variables['elements']['#entity_type'];
+  $view_mode = $variables['elements']['#view_mode'];
+  $hook = $variables['theme_hook_original'];
+
+  // Add a suggestion based on the entity type.
+  $suggestions[] = $hook . '__' . $entity_type;
```

Comment:
Only "nit" would be to explain someone could theme events by:

`civicrm-entity--civicrm-event.html.twig`

Reply:
But maybe that belongs in the twig file.

Reply:
Theme file makes sense to me, that's where the suggestions have been commented in D7 templates...

Reply:
Hmm I found a bug where civicrm_contact does not have the `#entity_type` key. We might need to loop through `$variables['elements']` and find the `CivicrmEntity`. Or maybe there's a better way?

<img width="500" alt="screenshot" src="https://user-images.githubusercontent.com/124599/43904562-03189a0a-9c00-11e8-8d20-c737664be8b5.png">

Reply:
The commit below fixes this.